File: UI/LatestNGApp/PMSUIApp/src/app/Services/task.service.ts
import { Injectable } from '@angular/core';
import { BehaviorSubject, Observable, of } from 'rxjs';
import { ProjectService } from './project.service';
import { HttpClient } from '@angular/common/http';
import { Projecttaskcount } from '../Models/projecttaskcount';
import { Mytasklist } from '../Models/mytasklist';
import { Taskdetail } from '../Models/taskdetail';
import { Moretaskdetail } from '../Models/moretaskdetail';
import { Alltasklist } from '../Models/alltasklist';
import { Taskidwithtitle } from '../Models/taskidwithtitle';
import { Addtask } from '../Models/addtask';

@Injectable({
  providedIn: 'root',
})
export class TaskService {

  constructor(private projectService: ProjectService,private httpClient:HttpClient) {}
  private selectedTaskIdSubject = new BehaviorSubject<any>(null);
  selectedTaskId$ = this.selectedTaskIdSubject.asObservable();

  setSelectedTaskId(id: number | null) {
    this.selectedTaskIdSubject.next(id);
  }
 
  GetProjectTaskCount(projectId:number):Observable<Projecttaskcount>{
    let url="http://localhost:5283/api/tasks/count/" +projectId
    return this.httpClient.get<Projecttaskcount>(url)
  } 
  
  GetMyTaskList(teamMemberId:number,timePeriod:string):Observable<Mytasklist[]>{
    let url="http://localhost:5283/api/tasks/mytasks/" +teamMemberId +"/" +timePeriod
    return this.httpClient.get<Mytasklist[]>(url)
  }
  getTaskDetails(taskId:number):Observable<Taskdetail>{
    let url="http://localhost:5283/api/tasks/taskDetail/" +taskId
    return this.httpClient.get<Taskdetail>(url)
  }
  getMoreTaskDetails(taskId:number):Observable<Moretaskdetail>{
    let url="http://localhost:5283/api/tasks/moretaskDetail/" +taskId
    return this.httpClient.get<Moretaskdetail>(url)
  }
  getAllTaskList(employeeId:number,timePeriod:string):Observable<Alltasklist[]>{
    let url="http://localhost:5283/api/tasks/alltasks/" + employeeId + "/" +timePeriod
    return this.httpClient.get<Alltasklist[]>(url)
  }
  getTaskIdWithTitle(employeeId :number,projectId :number,status :string):Observable<Taskidwithtitle[]>{
    let url="http://localhost:5283/api/tasks/tasktitle/ " + employeeId + "/" + projectId + "/" + status
    return this.httpClient.get<Taskidwithtitle[]>(url)
  }

  addTask(task:Addtask):Observable<boolean>{
    let url="http://localhost:5283/api/tasks/addtask"
    return this.httpClient.post<boolean>(url,task)
  }


}

File: API/Tasks/Services/TaskService.cs
using Transflower.PMSApp.Tasks.Services.Interfaces;
using Transflower.PMSApp.Tasks.Models;
using Transflower.PMSApp.Tasks.Repositories.Interfaces;
namespace Transflower.PMSApp.Tasks.Services;
public class TaskService:ITaskService
{
    private readonly ITaskRepository _taskRepository;
    public TaskService(ITaskRepository taskRepository){
        _taskRepository=taskRepository;
    }
    public async Task<ProjectTaskCount> GetProjectTaskCount(int projectId)=>await _taskRepository.GetProjectTaskCount(projectId);
    public async Task<List<MyTaskList>> GetMyTasksList(int teamMemberId,string timePeriod)=>await _taskRepository.GetMyTasksList(teamMemberId,timePeriod);

    public async Task<List<TaskResponse>> GetTasks(int projectId,int teamMemberId)=>await _taskRepository.GetTasks(projectId,teamMemberId);
    public async Task<TaskDetail> GetTaskDetail(int taskId)=>await _taskRepository.GetTaskDetail(taskId );
    public async Task<MoreTaskDetail> GetMoreTaskDetail(int taskId)=>await _taskRepository.GetMoreTaskDetail(taskId);
    public async Task<List<AllTaskList>> GetAllTaskList(int employeeId,string timePeriod)=>await _taskRepository.GetAllTaskList(employeeId,timePeriod);
    public async Task<List<TaskIdWithTitle>> GetTaskIdWithTitle(int employeeId,int projectId,string status)=>await _taskRepository.GetTaskIdWithTitle(employeeId,projectId,status);
    public async Task<bool> AddTask(Transflower.PMSApp.Tasks.Entities.Task task)=>await _taskRepository.AddTask(task);
    public async Task<Transflower.PMSApp.Tasks.Entities.Task> GetDetailsById(int taskId)=>await _taskRepository.GetDetailsById(taskId);
    public async Task<bool> UpdateTaskStatus(int taskId,string updateStatus)=>await _taskRepository.UpdateTaskStatus(taskId,updateStatus);




}